plutus-core/plutus-ir/test/transform/inline/letTypeAppMulti.golden

```diff
@@ -3,5 +3,8 @@
   (termbind
     (strict) (vardecl idFun (all a (type) (fun a a))) (abs a (type) (lam x a x))
   )
-  [ [ (builtin addInteger) (con integer 3) ] (con integer 3) ]
+  [
+    [ (builtin addInteger) [ (lam x (con integer) x) (con integer 3) ] ]
+    [ (lam x (con integer) x) (con integer 3) ]
+  ]
```

Comment:
🆗

Reply:
it seems somehow the applications get reduced further in most cases before the code change. I'm not sure why it is the case but the simplification pass should take care of that.